src/components/TheTable.vue

```diff
@@ -1,25 +1,20 @@
 <script setup lang="ts">
 import { Ref, computed, onMounted, ref } from "vue";
+import { BASE_URL, Todo } from "../main";
 import axios from "axios";
-
-const BASE_URL = "https://calm-plum-jaguar-tutu.cyclic.app/todos";
-
-type Todo = {
-  createdAt: string;
-  isComplete: boolean;
-  todoName: string;
-  updatedAt: string;
-  _id: string;
-};
+import { useRouter } from "vue-router";
+import TheLoader from "./TheLoader.vue";
 
 const props = defineProps<{
   titleFilter: string;
   statusFilter: "all" | "complete" | "incomplete";
 }>();
 
-const todos: Ref<Todo[]> = ref([]);
-const editingTodo: Ref<Todo | null> = ref(null);
-const addingTodo: Ref<string | null> = ref(null);
+const router = useRouter();
+
+const todos = ref<Todo[]>([]);
+const status = ref<"" | "loading" | "deleting">("loading");
```

Comment:
I think two boolean refs here like `isLoading` and `isDeleting` could also work or maybe, to be more flexible and also because really in both situations you are loading, a single ref for `isLoading` and a `message` ref to fill in a message from any function call that way if you wanted you could also display "Fetching todos..." for example without adding any extra code

Reply:
I don't think this kind of thing is bad generally, just maybe not the best solution in this case, and the empty string should maybe be something more meaningful